### ISSUE TITLE
Cannot load french ROM anymore

### ISSUE THREAD

**Issue by odrevet** [lordmir/landstalker_gfx#6]
Loading the french version of the ROM causes a SIGSEV. 

For information, The md5 of the ROM is 0bee733207842672479cc0f2775290ba
The backtrace is as follow: 

```
Thread 1 "landstalker_gfx" received signal SIGSEGV, Segmentation fault.
--Type <RET> for more, q to quit, c to continue without paging--
0x000055555558fdac in std::vector<unsigned int, std::allocator<unsigned int> >::size (this=0x100000001) at /usr/include/c++/10/bits/stl_vector.h:919
919	      { return size_type(this->_M_impl._M_finish - this->_M_impl._M_start); }
(gdb) bt
#0  0x000055555558fdac in std::vector<unsigned int, std::allocator<unsigned int> >::size() const (this=0x100000001) at /usr/include/c++/10/bits/stl_vector.h:919
#1  0x00005555555a1afd in SpriteGraphic::GetFrameCount(unsigned long) const (this=0x55555608fe70, animation=0) at ./SpriteGraphic.cpp:25
#2  0x0000555555584921 in MainFrame::OpenRomFile(wxString const&) (this=0x555555a94000, path=...) at ./MainFrame.cpp:175
#3  0x0000555555589065 in MainFrame::OnOpen(wxCommandEvent&) (this=0x555555a94000, event=...) at ./MainFrame.cpp:566
#4  0x00007ffff73d1661 in wxEvtHandler::ProcessEventIfMatchesId(wxEventTableEntryBase const&, wxEvtHandler*, wxEvent&) () at /lib/x86_64-linux-gnu/libwx_baseu-3.0.so.0
#5  0x00007ffff73d19fa in wxEvtHandler::SearchDynamicEventTable(wxEvent&) () at /lib/x86_64-linux-gnu/libwx_baseu-3.0.so.0
#6  0x00007ffff73d1a94 in wxEvtHandler::TryHereOnly(wxEvent&) () at /lib/x86_64-linux-gnu/libwx_baseu-3.0.so.0
#7  0x00007ffff73d1923 in wxEvtHandler::DoTryChain(wxEvent&) () at /lib/x86_64-linux-gnu/libwx_baseu-3.0.so.0
#8  0x00007ffff73d1bf1 in wxEvtHandler::ProcessEvent(wxEvent&) () at /lib/x86_64-linux-gnu/libwx_baseu-3.0.so.0
#9  0x00007ffff790a57a in wxWindowBase::TryAfter(wxEvent&) () at /lib/x86_64-linux-gnu/libwx_gtk3u_core-3.0.so.0
#10 0x00007ffff73d197b in wxEvtHandler::SafelyProcessEvent(wxEvent&) () at /lib/x86_64-linux-gnu/libwx_baseu-3.0.so.0
#11 0x00007ffff78bb0fe in wxMenuBase::SendEvent(int, int) () at /lib/x86_64-linux-gnu/libwx_gtk3u_core-3.0.so.0
#12 0x00007ffff77a2f43 in  () at /lib/x86_64-linux-gnu/libwx_gtk3u_core-3.0.so.0
#13 0x00007ffff618abcf in g_closure_invoke () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#14 0x00007ffff61b492b in  () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#15 0x00007ffff61a8748 in g_signal_emit_valist () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#16 0x00007ffff61a89a3 in g_signal_emit () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#17 0x00007ffff667295c in gtk_widget_activate () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#18 0x00007ffff653ef5e in gtk_menu_shell_activate_item () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#19 0x00007ffff653f233 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#20 0x00007ffff66c1308 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#21 0x00007ffff61a8834 in g_signal_emit_valist () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#22 0x00007ffff61a89a3 in g_signal_emit () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#23 0x00007ffff66884c4 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#24 0x00007ffff6529bc0 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#25 0x00007ffff652ae7b in gtk_main_do_event () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#26 0x00007ffff620b753 in  () at /lib/x86_64-linux-gnu/libgdk-3.so.0
#27 0x00007ffff6242e76 in  () at /lib/x86_64-linux-gnu/libgdk-3.so.0
#28 0x00007ffff5ee8bfb in g_main_context_dispatch () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#29 0x00007ffff5f3c468 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#30 0x00007ffff5ee8163 in g_main_loop_run () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#31 0x00007ffff65211cd in gtk_main () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#32 0x00007ffff7720bf5 in wxGUIEventLoop::DoRun() () at /lib/x86_64-linux-gnu/libwx_gtk3u_core-3.0.so.0
#33 0x00007ffff7285d41 in wxEventLoopBase::Run() () at /lib/x86_64-linux-gnu/libwx_baseu-3.0.so.0
#34 0x00007ffff724d3da in wxAppConsoleBase::MainLoop() () at /lib/x86_64-linux-gnu/libwx_baseu-3.0.so.0
#35 0x00007ffff72d973d in wxEntry(int&, wchar_t**) () at /lib/x86_64-linux-gnu/libwx_baseu-3.0.so.0
#36 0x00005555555827a1 in main(int, char**) (argc=1, argv=0x7fffffffdf38) at ./main.cpp:32
```

**Comment by odrevet** [comment]
The problem is related to Sprite Graphics (as hinted by the backtrace) 

The ROM  load when palettes are not set on sprites : In `MainFrame::OpenRomFile` commented 
* line 156 m_sprites[m_rom.read<uint8_t>(offset)].SetHighPalette(m_rom.read<uint8_t>(offset + 1) & 0x7F); 
* line 160 m_sprites[m_rom.read<uint8_t>(offset)].SetLowPalette(m_rom.read<uint8_t>(offset + 1));

**Comment by lordmir** [comment]
The issue is caused by the hex offsets of certain data structures (including the sprite palettes) being different across the different ROM versions.
I have moved the ROM offsets to a common location and have added code to detect the ROM version and select the appropriate offsets depending on the version of the ROM. Currently, this will result in the application reporting an error that the ROM version selected is not compatible.
Once the offsets for the different ROM versions have been added to the code, then this should allow viewing of the contents of all revisions of the ROM. Hopefully should have something together in the next few days.
